### PR TITLE
feat(sdk-coin-avaxp): add avaxp new SDK

### PR DESCRIPTION
**Pull request by noel-bitgo**
CR-1073

TICKET: CR-1073

### CODE REVIEW

**Review by sol-mocha — CHANGES_REQUESTED** [review]
https://github.com/BitGo/BitGoJS/pull/4119

**Review by zahin-mohammad — CHANGES_REQUESTED** [review]
Few comments:
- Browser tests won't pass due to cypress
- Changing min version to node 18 is a breaking change, you need to add the appropriate footer tag for this

I already did both in my node 16 PR, please rebate off of mine.
https://github.com/BitGo/BitGoJS/pull/4119

File: modules/sdk-coin-avaxp/src/lib/permissionlessValidatorTxBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    this.validateNodeID(value);
+    this._nodeID = value;
+    return this;
+  }
+
+  /**
+   *
+   * @param blsKey
+   */
+  blsKey(blsKey: string): this {
```
Comment (noel-bitgo) [inline modules/sdk-coin-avaxp/src/lib/permissionlessValidatorTxBuilder.ts:56]: oh i used `blsPublicKey`

File: modules/sdk-coin-avaxp/src/lib/permissionlessValidatorTxBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
+import { BuildTransactionError, TransactionType } from '@bitgo/sdk-core';
+import { AvalancheNetwork, BaseCoin as CoinConfig } from '@bitgo/statics';
+import { BinTools, BN } from 'avalanche';
+import { AddPermissionlessValidatorTx } from 'bitgo-aaron-avalanchejs/dist/serializable/pvm';
+import { Tx } from './iface';
+
+export class PermissionlessValidatorTxBuilder {
```
Comment (noel-bitgo) [inline modules/sdk-coin-avaxp/src/lib/permissionlessValidatorTxBuilder.ts:7]: add [getPermissionlessValidatorTxBuilder](https://github.com/BitGo/BitGoJS/blob/e441e877057c8ba4f22d1b721423c0cbebc7bb3b/modules/sdk-coin-avaxp/src/lib/transactionBuilderFactory.ts#L77)

File: modules/sdk-coin-avaxp/package.json [hunk 3]
```diff
@@ -51,6 +51,7 @@
     "@bitgo/sdk-core": "^26.3.0",
     "@bitgo/statics": "^48.1.0",
     "@bitgo/utxo-lib": "^9.34.0",
+    "bitgo-aaron-avalanchejs": "^4.0.5",
```
Comment (zahin-mohammad) [inline modules/sdk-coin-avaxp/package.json:54]: If we know this is a working fix, can we fork this repo into a bitgo controlled repo?

Comment (noel-bitgo) [reply modules/sdk-coin-avaxp/package.json:54]: yes on it

Comment (andrew-scott-fischer) [inline modules/sdk-coin-avaxp/package.json:54]: This needs to be 
```suggestion
    "@bitgo/avalanchejs": "1.0.0-alpha.3",
```

Comment (DinshawKothari) [reply modules/sdk-coin-avaxp/package.json:54]: Thanks @andrew-scott-fischer. Will use this in PR - https://github.com/BitGo/BitGoJS/pull/4366 too.

**Comment by andrew-scott-fischer** [comment]
> > Few comments:
> > 
> > * Browser tests won't pass due to cypress

Looks like `browser-test`s did pass? Why wasn't that expected? @zahin-mohammad @noel-bitgo ?

**Comment by zahin-mohammad** [comment]
> > > Few comments:
> > > 
> > > * Browser tests won't pass due to cypress
> 
> Looks like `browser-test`s [did pass](https://github.com/BitGo/BitGoJS/actions/runs/8440139442/job/23116363822?pr=4361)? Why wasn't that expected? @zahin-mohammad @noel-bitgo ?

Honestly not 100% sure why, but upgrading cypress fixed it for me. https://bitgo.slack.com/archives/C010AEXLLCR/p1711229931703669

**Comment by andrew-scott-fischer** [comment]
> * Changing min version to node 18 is a breaking change, you need to add the appropriate footer tag for this

@zahin-mohammad When you merge your PR to remove 16, [update the readme too](https://github.com/BitGo/BitGoJS/blob/master/README.md?plain=1#L64C1-L64C7)

**Comment by zahin-mohammad** [comment]
> > * Changing min version to node 18 is a breaking change, you need to add the appropriate footer tag for this
> 
> @zahin-mohammad When you merge your PR to remove 16, [update the readme too](https://github.com/BitGo/BitGoJS/blob/master/README.md?plain=1&rgh-link-date=2024-03-26T17%3A35%3A30Z#L64C1-L64C7)

Ah must have missed it, good catch.

**Comment by zahin-mohammad** [comment]
> > * Changing min version to node 18 is a breaking change, you need to add the appropriate footer tag for this
> 
> @zahin-mohammad When you merge your PR to remove 16, [update the readme too](https://github.com/BitGo/BitGoJS/blob/master/README.md?plain=1&rgh-link-date=2024-03-26T17%3A35%3A30Z#L64C1-L64C7)

https://github.com/BitGo/BitGoJS/pull/4371/files

**Review by zahin-mohammad — APPROVED** [review]

**Review by ewangbitgo — APPROVED** [review]

**Review by YPAzevedo — APPROVED** [review]

**Review by OttoAllmendinger — APPROVED** [review]

**Review by davidkaplanbitgo — APPROVED** [review]

**Review by CapnMigraine — APPROVED** [review]

**Pull request merged**